### ISSUE TITLE
Bug report - hitting enter in an editable table cell

### ISSUE THREAD

**Issue by neterium** [primefaces/primeng#3542]
Hi, when I hit enter in an editable cell (4.1.2) i get the following stack trace:

RROR TypeError: Cannot read property 'classList' of null
    at DomHandler.webpackJsonp.../../../../primeng/components/dom/domhandler.js.DomHandler.removeClass (domhandler.js:35)
    at DataTable.webpackJsonp.../../../../primeng/components/datatable/datatable.js.DataTable.switchCellToViewMode (datatable.js:1305)
    at DataTable.webpackJsonp.../../../../primeng/components/datatable/datatable.js.DataTable.onCellEditorKeydown (datatable.js:1339)
    at Object.eval [as handleEvent] (TableBody.html:31)
    at handleEvent (core.es5.js:11997)
    at callWithDebugContext (core.es5.js:13458)
    at Object.debugHandleEvent [as handleEvent] (core.es5.js:13046)
    at dispatchEvent (core.es5.js:8601)
    at core.es5.js:9212
    at HTMLInputElement.<anonymous> (platform-browser.es5.js:2651)

I checked the code and found the following, looks like the editingCell is set to null then passed (as null) to the domHandler. I suppose it's not what was supposed to be done :)

**DataTable.prototype.switchCellToViewMode = function (element) {
        this.editingCell = null;
        var cell = this.findCell(element);
        this.domHandler.removeClass(this.editingCell, 'ui-cell-editing');
        this.unbindDocumentEditListener();
    };**

**Comment by badpenguin** [comment]
I have the same problem, also if pressing ESC or Enter or whatever. The cells remains stuck in "edit mode".
```
TableBody.html:30 ERROR CONTEXT DebugContext_ {view: Object, nodeIndex: 0, nodeDef: Object, elDef: Object, elView: Object}
View_TableBody_11 @ TableBody.html:30
webpackJsonp.../../../core/@angular/core.es5.js.DebugContext_.logError @ core.es5.js:13398
webpackJsonp.../../../core/@angular/core.es5.js.ErrorHandler.handleError @ core.es5.js:1085
(anonymous) @ core.es5.js:9216
(anonymous) @ platform-browser.es5.js:2651
webpackJsonp.../../../../zone.js/dist/zone.js.ZoneDelegate.invokeTask @ zone.js:424
onInvokeTask @ core.es5.js:3881
webpackJsonp.../../../../zone.js/dist/zone.js.ZoneDelegate.invokeTask @ zone.js:423
webpackJsonp.../../../../zone.js/dist/zone.js.Zone.runTask @ zone.js:191
webpackJsonp.../../../../zone.js/dist/zone.js.ZoneTask.invokeTask @ zone.js:498
invokeTask @ zone.js:1370
globalZoneAwareCallback @ zone.js:1388
```

my setup:

```dependencies": {
    "@angular/animations": "^4.3.2",
    "@angular/common": "^4.0.0",
    "@angular/compiler": "^4.0.0",
    "@angular/core": "^4.0.0",
    "@angular/forms": "^4.0.0",
    "@angular/http": "^4.0.0",
    "@angular/platform-browser": "^4.0.0",
    "@angular/platform-browser-dynamic": "^4.0.0",
    "@angular/router": "^4.0.0",
    "angularfire2": "^4.0.0-rc.1",
    "core-js": "^2.4.1",
    "firebase": "^4.2.0",
    "font-awesome": "^4.7.0",
    "primeng": "^4.1.2",
    "rxjs": "^5.4.1",
    "zone.js": "^0.8.14"
  },
  "devDependencies": {
    "@angular/cli": "1.2.6",
    "@angular/compiler-cli": "^4.0.0",
    "@angular/language-service": "^4.0.0",
    "@types/jasmine": "~2.5.53",
    "@types/jasminewd2": "~2.0.2",
    "@types/node": "~6.0.60",
    "codelyzer": "~3.0.1",
    "jasmine-core": "~2.6.2",
    "jasmine-spec-reporter": "~4.1.0",
    "karma": "~1.7.0",
    "karma-chrome-launcher": "~2.1.1",
    "karma-cli": "~1.0.1",
    "karma-coverage-istanbul-reporter": "^1.2.1",
    "karma-jasmine": "~1.1.0",
    "karma-jasmine-html-reporter": "^0.2.2",
    "protractor": "~5.1.2",
    "ts-node": "~3.0.4",
    "tslint": "~5.3.2",
    "typescript": "~2.3.3"
  }
```

**Comment by PaulAGH** [comment]
Same problem here also.  To repeat do following.  (Problem is not present in 4.1.1.)

Pull down 4.1.2 tag of PrimeNG
npm install
npm start

Navigate to DataTable, and then click on Editable, and then click on a cell in the editable table. Press return and observe the following error in the Chrome console. (This does not currently occur in the online demo at https://www.primefaces.org/primeng/#/datatable/editable  - perhaps online demo is not using the most current version of PrimeNG?)

TableBody.html:30 ERROR TypeError: Cannot read property 'classList' of null
at DomHandler.webpackJsonp../src/app/components/dom/domhandler.ts.DomHandler.removeClass (domhandler.ts?017d:34)
at ...

This was observed on Windows 7 with the following packages installed

npm list --depth=0
primeng@4.1.2 E:\Software\Javascript\primeng
+-- @angular/animations@4.3.3
+-- @angular/cli@1.1.0
+-- @angular/common@4.3.3
+-- @angular/compiler@4.3.3
+-- @angular/compiler-cli@4.3.3
+-- @angular/core@4.3.3
+-- @angular/forms@4.3.3
+-- @angular/http@4.3.3
+-- @angular/language-service@4.3.3
+-- @angular/platform-browser@4.3.3
+-- @angular/platform-browser-dynamic@4.3.3
+-- @angular/router@4.3.3
+-- @types/jasmine@2.5.45
+-- @types/node@6.0.85
+-- chart.js@2.1.3
+-- codelyzer@3.0.1
+-- core-js@2.4.1
+-- del@2.2.2
+-- font-awesome@4.7.0
+-- fullcalendar@3.4.0
+-- gulp@3.9.1
+-- gulp-concat@2.6.1
+-- gulp-flatten@0.2.0
+-- gulp-rename@1.2.2
+-- gulp-uglify@1.5.4
+-- gulp-uglifycss@1.0.8
+-- install@0.10.1
+-- intl@1.2.5
+-- jasmine-core@2.6.4
+-- jasmine-spec-reporter@4.1.1
+-- jquery@3.2.1
+-- karma@1.7.0
+-- karma-chrome-launcher@2.1.1
+-- karma-cli@1.0.1
+-- karma-coverage-istanbul-reporter@1.3.0
+-- karma-jasmine@1.1.0
+-- karma-jasmine-html-reporter@0.2.2
+-- moment@2.18.1
+-- npm@5.3.0
+-- prismjs@1.6.0
+-- protractor@5.1.2
+-- quill@1.3.0
+-- rxjs@5.4.2
+-- ts-node@3.0.6
+-- tslint@5.3.2
+-- typescript@2.3.4
+-- web-animations-js@2.3.1
`-- zone.js@0.8.16

**Comment by patriknil90** [comment]
Same problem here!

Tested in Chrome Version 59.0.3071.115.

Setup:
   ```
 "@angular/animations": "^4.3.3",
    "@angular/common": "^4.3.3",
    "@angular/compiler": "^4.3.3",
    "@angular/core": "^4.3.3",
    "@angular/forms": "^4.3.3",
    "@angular/http": "^4.3.3",
    "@angular/material": "^2.0.0-beta.5",
    "@angular/platform-browser": "^4.3.3",
    "@angular/platform-browser-dynamic": "^4.3.3",
    "@angular/router": "^4.3.3",
    "@ng-idle/core": "^2.0.0-beta.12",
    "@ng-idle/keepalive": "^2.0.0-beta.12",
    "angular2-image-gallery": "^1.0.1",
    "angular2-moment": "^1.2.0",
    "apollo-angular": "^0.13.0",
    "apollo-client": "^1.9.0",
    "apollo-client-rxjs": "^0.5.2",
    "chart.js": "^2.4.0",
    "classlist.js": "^1.1.20150312",
    "core-js": "^2.4.1",
    "graphql": "^0.10.1",
    "graphql-tag": "^2.4.2",
    "hammerjs": "^2.0.8",
    "highcharts": "^5.0.10",
    "intl": "^1.2.5",
    "ng2-highcharts": "^1.0.0",
    "ng2-pdf-viewer": "^1.1.0",
    "ngx-infinite-scroll": "^0.4.1",
    "ngx-tooltip": "0.0.9",
    "primeng": "^4.1.2",
    "redux": "^3.5.2",
    "rxjs": "^5.4.2",
    "web-animations-js": "^2.2.5",
    "zone.js": "^0.8.4"
  },
  "devDependencies": {
    "@angular/cli": "^1.2.7",
    "@types/highcharts": "^4.2.54",
    "@types/jasmine": "2.5.38",
    "@types/node": "~6.0.60",
    "codelyzer": "~2.0.0",
    "jasmine-core": "~2.5.2",
    "jasmine-spec-reporter": "~3.2.0",
    "karma": "~1.4.1",
    "karma-chrome-launcher": "~2.0.0",
    "karma-cli": "~1.0.1",
    "karma-coverage-istanbul-reporter": "^0.2.0",
    "karma-jasmine": "~1.1.0",
    "karma-jasmine-html-reporter": "^0.2.2",
    "protractor": "~5.1.0",
    "ts-node": "~2.0.0",
    "tslint": "~4.5.0",
    "typescript": "^2.4.2"

```

**Comment by cagataycivici** [comment]
Fixed via #3553
